Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelWizard/ScanAPIModelWizardPage.xaml.cs

```diff
@@ -91,15 +122,77 @@ public void WizardEvent(WizardEventArgs WizardEventArgs)
                     Parse();
                 }
             }
+            else if (WizardEventArgs.EventType == EventType.LeavingForNextPage)
+            {
+                if (AddAPIModelWizard.DeltaModelsList != null && AddAPIModelWizard.DeltaModelsList.Count > 0)
+                {
+                    bool mergeIssue = false;
+                    bool notifyReplaceAPI = false;
+                    AddAPIModelWizard.LearnedAPIModelsList.Clear();
+                    foreach (DeltaAPIModel deltaModel in AddAPIModelWizard.DeltaModelsList.Where(m => m.IsSelected == true))
+                    {
+                        ApplicationAPIModel selectedAPIModel = null;
+
+                        if (deltaModel.DefaultOperationEnum == DeltaAPIModel.eHandlingOperations.Add)
+                            selectedAPIModel = deltaModel.learnedAPI;
+                        else if (deltaModel.DefaultOperationEnum == DeltaAPIModel.eHandlingOperations.MergeChanges)
+                        {
+                            if (deltaModel.MergedAPIModel == null)
+                            {
+                                mergeIssue = true;
+                                break;
+                            }
+                            else
+                            {
+                                deltaModel.MergedAPIModel.ContainingFolder = deltaModel.matchingAPIModel.ContainingFolderFullPath;
+                                deltaModel.MergedAPIModel.Guid = deltaModel.matchingAPIModel.Guid;
```

Comment:
you probably will need to do it (copy the GUID and rest) only on finish because what will happen in below scanrio:
1- user had a match 
2- user selected replace existing
3- user clicked next and this code was copying the GUI
4- user clicked back and selected "Add New"
so now we will keep the existing and will add new but both will be with same GUID, no?

Reply:
Updated !